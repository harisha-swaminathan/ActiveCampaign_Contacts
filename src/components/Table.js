/* eslint-disable react/prop-types */
import Contact from './Contact';
import React from 'react';

function Table(props){
    return(
     <div className="table">
        {/* Setting all input values to readonly inorder to avoid the "add onchange handler" warning */}
        <div className="table__filterBar">
            <div>
                <button className="filterBar__button">Edit</button>
                <button className="filterBar__button">More...</button>
            </div>
           <div>
           <select className="filterBar__dropdown" value="tags">
                <option selected>All tags</option>
            </select>
            <select className="filterBar__dropdown" value="locations" readOnly>
                <option selected>All locations</option>
            </select>
            <input className="text-input"
              type="text"
              placeholder="Search"
              readOnly
            />
           </div>
        </div>
        {/* <div className="table__header">
            <input type="checkbox"/>
            <span>Contact</span>
            <span>Total Value</span>
            <span>Location</span>
            <span>Deals</span>
            <span>Tags</span>
        </div>
        <div className="table__rows">
            {props.contacts.map((contact)=>{
                return <div className="table__row">
                    <Contact key={contact.id} contact = {contact}/>
                </div>
            })}
        </div> */}
        <table>
            <tr className="table__header">
                <input type="checkbox"/> 
                <th>Contact</th>
                <th>Total Value</th>
                <th>Location</th>
                <th>Deals</th>
                <th>Tags</th>
            </tr>
            {props.contacts.map((contact)=>{
                return <tr key={contact.id}>
                    <Contact key={contact.id} contact = {contact}/>
                </tr>
            })}
        </table>
     </div>
    );
}

export default Table;






