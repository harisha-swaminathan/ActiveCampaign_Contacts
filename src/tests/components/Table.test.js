import React from 'react';
import { shallow } from 'enzyme';
import contacts from '../fixtures/contacts';
import Table from '../../components/Table';

test('should render ExpenseDashboardPage correctly', () => {
  const wrapper = shallow(<Table contacts={contacts.data.contacts}/>);
  expect(wrapper).toMatchSnapshot();
});
