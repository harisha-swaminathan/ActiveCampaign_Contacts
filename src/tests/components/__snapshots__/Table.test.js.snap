// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseDashboardPage correctly 1`] = `
<div
  className="table"
>
  <div
    className="table__filterBar"
  >
    <div>
      <button
        className="filterBar__button"
        type="button"
      >
        Edit
      </button>
      <button
        className="filterBar__button"
        type="button"
      >
        More...
      </button>
    </div>
    <div>
      <select
        className="filterBar__dropdown"
        defaultValue="All tags"
      >
        <option>
          All tags
        </option>
      </select>
      <select
        className="filterBar__dropdown"
        defaultValue="All locations"
      >
        <option>
          All locations
        </option>
      </select>
      <input
        className="filterBar__search"
        placeholder="Search"
        readOnly={true}
        type="text"
      />
    </div>
  </div>
  <table>
    <thead>
      <tr
        className="table__header"
      >
        <th>
          <input
            type="checkbox"
          />
        </th>
        <th>
          Contact
        </th>
        <th>
          Total Value
        </th>
        <th>
          Location
        </th>
        <th>
          Deals
        </th>
        <th>
          Tags
        </th>
      </tr>
    </thead>
    <Contacts />
  </table>
</div>
`;
